### ISSUE TITLE
Add NodeSelector to Deployments, StatefulSets, DaemonSets so that pods don't end up on windows nodes when running a hybrid cluster

### ISSUE THREAD

**Issue by ffd2subroutine** [coreos/prometheus-operator#1244]
**What did you do?**
```
cd contrib/kube-prometheus/
./hack/cluster-monitoring/deploy
```

**What did you expect to see?**
pods deployed on linux nodes

**What did you see instead? Under which circumstances?**
alertmanager-main and prometheus-k8s StatefulSets are deployed on windows node.

NOTE:
I'm able to edit deployment and daemonset templates under **/contrib/kube-prometheus/manifests/** and add
```
nodeSelector:
  beta.kubernetes.io/os: linux
```
for grafana, kube-state-metrics, node-exporter and prometheus-operator, so these get scheduled on linux nodes. 
I need same for alertmanager-main and prometheus-k8s statefulsets as well. Also, for some reason I'm not able to edit these two resources after I deploy them, was trying to add nodeSelector but it fails, for example:
```
kubectl get statefulset alertmanager-main -n monitoring -o yaml > alertmanager-statefulset.yaml
```
add nodeSelector then:
```
kubectl apply -f alertmanager-statefulset.yaml
```
**Environment**

* Kubernetes version information:
```
Client Version: version.Info{Major:"1", Minor:"9", GitVersion:"v1.9.3", GitCommit:"d2835416544f298c919e2ead3be3d0864b52323b", GitTreeState:"clean", BuildDate:"2018-02-07T12:22:21Z", GoVersion:"go1.9.2", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"9", GitVersion:"v1.9.3", GitCommit:"d2835416544f298c919e2ead3be3d0864b52323b", GitTreeState:"clean", BuildDate:"2018-02-07T11:55:20Z", GoVersion:"go1.9.2", Compiler:"gc", Platform:"linux/amd64"}
```
* Kubernetes cluster kind: 

	ace-engine on Azure

**Comment by ffd2subroutine** [comment]
Hi @brancz, yeah it works, thank you!
Anyway, it would be nice to have this by default, I can make a pull request if you are interested.

**Comment by brancz** [comment]
Technically Prometheus is able to build on windows and same for all the other components, so ideally those labels would be in the manifests, not in the operator go code. That would be similar to adding tolerations as I demonstrated how to do here: https://github.com/coreos/prometheus-operator/pull/1242#issuecomment-383477997.

**Comment by ffd2subroutine** [comment]
Yeah I meant updating manifests files under /contrib/kube-prometheus/manifests, not touching the operator's code, until prometheus-operator is ready for windows clusters as well.

**Comment by brancz** [comment]
You are likely using an outdated commit, on latest master, the `hack/cluster-monitoring/deploy` script does not exist anymore and the manifests all have linux node selectors. Could you try latest master?
